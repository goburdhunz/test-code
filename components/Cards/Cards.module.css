.card {
    display: flex;
    flex-direction: column;
    margin: 1rem;
    padding: 0.8rem;
    text-align: left;
    color: white;
    background-color: black;
    text-decoration: none;
    border: 1px solid #eaeaea;
    border-radius: 0.625rem;
    transition: color 0.15s ease, border-color 0.15s ease;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: min-content;
    height: 25rem;
    justify-content: space-evenly;
    font-weight: 700;
    font-family: Arial, Helvetica, sans-serif;
}

.card img {
    width: 4.375rem;
    height: 4.375rem;
}


.card:hover,
.card:focus,
.card:active {
    border-color: #0070f3;
}