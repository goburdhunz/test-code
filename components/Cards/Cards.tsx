import React from 'react';
import { LaunchResponseData } from '../../types/types'
import Image from 'next/image';
import styles from './Cards.module.css';

const Cards = ({ name, date_utc, cores, payloads, success, failures, links: { patch: { small } } }: LaunchResponseData) => {
    const formattedUtcDate = date_utc?.split('T')?.[0]
    const showFailure = success ? "" : `Reason for Failure:${failures?.[0]?.reason}`
    return (
        <div className={styles.card}>
            <Image priority src={small} alt="rocketLogo" width={70} height={70} />
            <span>Name: {name}</span>
            <span>Core Serial: {cores?.[0]?.core}</span>
            <span>Date: {formattedUtcDate}</span>
            <span>Payload Id: {payloads?.[0]?.id}</span>
            <span>Payload type: {payloads?.[0]?.type}</span>
            <span>Successful Launch: {success ? 'Yes' : 'No'}</span>
            <span>{showFailure}</span>
        </div>
    )
}

export default Cards