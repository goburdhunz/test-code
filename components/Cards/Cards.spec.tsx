import Cards from './Cards';
import { render } from '@testing-library/react'

jest.mock('./Cards.module.css')

const props = {
    name: 'falcon8',
    date_utc: '2006-03-24T22:30:00.000Z',
    cores: [{
        core: "5e9e289df35918033d3b2623"
    }],
    payloads: [{ id: "5eb0e4b5b6c3bb0006eeb1e1", type: "Satellite" }],
    success: true,
    failures: undefined,
    links: { patch: { small: "https://images2.imgbox.com/94/f2/NN6Ph45r_o.png" } }
}

describe('Cards component', () => {
    it('should render correctly', async () => {
        const { container } = render(<Cards {...props} />)
        expect(container).toMatchSnapshot()
    })
})