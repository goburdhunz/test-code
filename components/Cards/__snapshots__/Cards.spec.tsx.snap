// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component should render correctly 1`] = `
<div>
  <div>
    <img
      alt="rocketLogo"
      data-nimg="1"
      decoding="async"
      height="70"
      src="/_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=256&q=75 2x"
      style="color: transparent;"
      width="70"
    />
    <span>
      Name: 
      falcon8
    </span>
    <span>
      Core Serial: 
      5e9e289df35918033d3b2623
    </span>
    <span>
      Date: 
      2006-03-24
    </span>
    <span>
      Payload Id: 
      5eb0e4b5b6c3bb0006eeb1e1
    </span>
    <span>
      Payload type: 
      Satellite
    </span>
    <span>
      Successful Launch: 
      Yes
    </span>
    <span />
  </div>
</div>
`;
